src/utils/ucx/ucx_utils.cpp

```diff
@@ -67,6 +81,42 @@ void ucx_modify_config(ucp_config_t *config, std::string_view key,
     }
 }
 
+[[nodiscard]] std::string_view
+to_string (ucp_err_handling_mode_t t) {
+    switch (t) {
+    case UCP_ERR_HANDLING_MODE_NONE:
+        return "none";
+    case UCP_ERR_HANDLING_MODE_PEER:
+        return "peer";
+    default:
+        throw std::invalid_argument (std::to_string (t));
+        ;
+    }
+}
+
+[[nodiscard]] ucp_err_handling_mode_t
+err_mode_from_string (const std::string &s) {
+    try {
+        for (const auto &mode : nixl_ucx_err_handling_modes) {
+            if (to_string (mode) == s) {
+                return mode;
+            }
+        }
+
+        throw std::invalid_argument (s);
+    }
+    catch (const std::invalid_argument &) {
+        std::string err_msg = "Invalid error handling mode: " + s + ". Valid values are: <";
+        for (const auto &mode : nixl_ucx_err_handling_modes) {
+            err_msg += std::string (to_string (mode)) + "|";
+        }
+
+        err_msg.pop_back();
+        err_msg += ">";
+        throw std::invalid_argument (err_msg);
+    }
```

Comment:
IMO this try-catch block is redundant.
In fact it's only handling exception from line 106, because exception from line 101 is not possible. And even if it happens - no need to catch it here
```suggestion
        for (const auto &mode : nixl_ucx_err_handling_modes) {
            if (to_string (mode) == s) {
                return mode;
            }
        }

        std::string err_msg = "Invalid error handling mode: " + s + ". Valid values are: <";
        for (const auto &mode : nixl_ucx_err_handling_modes) {
            err_msg += std::string (to_string (mode)) + "|";
        }

        err_msg.pop_back();
        err_msg += ">";
        throw std::invalid_argument (err_msg);
```